### ISSUE TITLE
DiT results on CIFAR10

### ISSUE THREAD

**Issue by yuanzhi-zhu** [facebookresearch/DiT#84]
have you tried to run DiT on CIFAR10 dataset?
I did some simple expr and found that DiT does not work well on CIFAR10.

**Comment by tanghengjian** [comment]
i also found sample.py script always give same result image under same label. 
In the workflow of DiTBlock, i wonder there is no cross attention , so i guess the variation ability may be a challange to DiT?

**Comment by yuanzhi-zhu** [comment]
> i also found sample.py script always give same result image under same label.
In the workflow of DiTBlock, i wonder there is no cross attention , so i guess the variation ability may be a challange to DiT?

Hi @tanghengjian, I do not know if your question is related to the cifar expr, but did you change the seed in the sample.py script?
https://github.com/facebookresearch/DiT/blob/ed81ce2229091fd4ecc9a223645f95cf379d582b/sample.py#L23

**Comment by tanghengjian** [comment]
run with default value. 
by the way, i found cifar10 dataset is  only 32*32 pixel with 10 classes, it means the y condition changes from 0 to 9. 
do you have tested the  mscoco  dataset  in DiT model  with label condition?

**Comment by zhengyu-su** [comment]
> run with default value. by the way, i found cifar10 dataset is only 32*32 pixel with 10 classes, it means the y condition changes from 0 to 9. do you have tested the mscoco dataset in DiT model with label condition?

How do you link CIFAR10 classes to the ImageNet 1k classes?

**Comment by forever208** [comment]
I trained the model on CIFAR-10, the FID is very high, about FID-50k = 10.

Regarding the results on CelebA 64x64 and FFHQ 128x128, refer to the paper:
https://arxiv.org/abs/2412.15032

**Comment by shuaikaishi** [comment]
> I trained the model on CIFAR-10, the FID is very high, about FID-50k = 10.
> 
> Regarding the results on CelebA 64x64 and FFHQ 128x128, refer to the paper: https://arxiv.org/abs/2412.15032

The cifar-10 contains 50k training images and 10k test images.    The "FID-50k" is evaluated on training image? Or 50k generated images and 10k test images?

**Comment by forever208** [comment]
> > I trained the model on CIFAR-10, the FID is very high, about FID-50k = 10.
> > Regarding the results on CelebA 64x64 and FFHQ 128x128, refer to the paper: https://arxiv.org/abs/2412.15032
> 
> The cifar-10 contains 50k training images and 10k test images. The "FID-50k" is evaluated on training image? Or 50k generated images and 10k test images?

evaluated on training images, the goal is to learn the data distribution from the training samples

**Comment by shuaikaishi** [comment]
> > > I trained the model on CIFAR-10, the FID is very high, about FID-50k = 10.
> > > Regarding the results on CelebA 64x64 and FFHQ 128x128, refer to the paper: https://arxiv.org/abs/2412.15032
> > 
> > 
> > The cifar-10 contains 50k training images and 10k test images. The "FID-50k" is evaluated on training image? Or 50k generated images and 10k test images?
> 
> evaluated on training images, the goal is to learn the data distribution from the training samples

Thank you for your reply. How many epochs are you trained? Was the batch size set to 256 as DiT trained on ImageNet? I can test it on cifar-10 later and share the results.

**Comment by forever208** [comment]
> > > > I trained the model on CIFAR-10, the FID is very high, about FID-50k = 10.
> > > > Regarding the results on CelebA 64x64 and FFHQ 128x128, refer to the paper: https://arxiv.org/abs/2412.15032
> > > 
> > > 
> > > The cifar-10 contains 50k training images and 10k test images. The "FID-50k" is evaluated on training image? Or 50k generated images and 10k test images?
> > 
> > 
> > evaluated on training images, the goal is to learn the data distribution from the training samples
> 
> Thank you for your reply. How many epochs are you trained? Was the batch size set to 256 as DiT trained on ImageNet? I can test it on cifar-10 later and share the results.

about 300k~500k steps with batch size 256

**Comment by shuaikaishi** [comment]
> > > > > I trained the model on CIFAR-10, the FID is very high, about FID-50k = 10.
> > > > > Regarding the results on CelebA 64x64 and FFHQ 128x128, refer to the paper: https://arxiv.org/abs/2412.15032
> > > > 
> > > > 
> > > > The cifar-10 contains 50k training images and 10k test images. The "FID-50k" is evaluated on training image? Or 50k generated images and 10k test images?
> > > 
> > > 
> > > evaluated on training images, the goal is to learn the data distribution from the training samples
> > 
> > 
> > Thank you for your reply. How many epochs are you trained? Was the batch size set to 256 as DiT trained on ImageNet? I can test it on cifar-10 later and share the results.
> 
> about 300k~500k steps with batch size 256

I tested the DiT-S/8 and DiT-B/8 on CIFAR10 for 350k training steps. Because I only have RTX 4090GPU. 
Fid of DiT S/8 is 157.01 and fid of DiT-B/8 is 149.6.
